cmd/backup-manager/app/util/util.go

```diff
@@ -161,70 +161,103 @@ func GetOptionValueFromEnv(option, envPrefix string) string {
+	args = append(args, constructBRGlobalOptions(config.BR)...)
 	storageArgs, err := getRemoteStorage(backup.Spec.StorageProvider)
 	if err != nil {
 		return nil, err
 	}
 	args = append(args, storageArgs...)
-	if (backup.Spec.Type == v1alpha1.BackupTypeDB || backup.Spec.Type == v1alpha1.BackupTypeTable) && config.DB != "" {
-		args = append(args, fmt.Sprintf("--db=%s", config.DB))
-	}
-	if backup.Spec.Type == v1alpha1.BackupTypeTable && config.Table != "" {
-		args = append(args, fmt.Sprintf("--table=%s", config.Table))
+
+	if config.TableFilter != nil && len(config.TableFilter) > 0 {
+		for _, tableFilter := range config.TableFilter {
+			args = append(args, "--filter", tableFilter)
+		}
+	} else {
+		if ((backup.Spec.Type == v1alpha1.BackupTypeDB || backup.Spec.Type == v1alpha1.BackupTypeTable) && config.BR.DB != "") ||
+			(backup.Spec.Type == v1alpha1.BackupTypeTable && config.BR.Table != "") {
+			if (backup.Spec.Type == v1alpha1.BackupTypeDB || backup.Spec.Type == v1alpha1.BackupTypeTable) && config.BR.DB != "" {
+				args = append(args, fmt.Sprintf("--db=%s", config.BR.DB))
+			}
+			if backup.Spec.Type == v1alpha1.BackupTypeTable && config.BR.Table != "" {
+				args = append(args, fmt.Sprintf("--table=%s", config.BR.Table))
+			}
+		} else {
+			args = append(args, defaultTableFilterOptions...)
+		}
 	}
+
 	return args, nil
 }
 
 // ConstructDumplingOptionsForBackup constructs dumpling options for backup
 func ConstructDumplingOptionsForBackup(backup *v1alpha1.Backup) []string {
 	var args []string
-	config := backup.Spec.Dumpling
-	if config == nil {
+	config := backup.Spec
+
+	if config.TableFilter != nil && len(config.TableFilter) > 0 {
+		for _, tableFilter := range config.TableFilter {
+			args = append(args, "--filter", tableFilter)
+		}
+	} else {
+		if config.Dumpling != nil && config.Dumpling.TableFilter != nil && len(config.Dumpling.TableFilter) > 0 {
+			for _, tableFilter := range config.Dumpling.TableFilter {
+				args = append(args, "--filter", tableFilter)
+			}
+		} else {
+			args = append(args, defaultTableFilterOptions...)
+		}
+	}
+
+	if config.Dumpling == nil {
 		args = append(args, defaultOptions...)
-		args = append(args, defaultTableFilterOptions...)
 		return args
 	}
 
-	if len(config.Options) != 0 {
-		args = append(args, config.Options...)
+	if len(config.Dumpling.Options) != 0 {
+		args = append(args, config.Dumpling.Options...)
 	} else {
 		args = append(args, defaultOptions...)
 	}
 
-	if len(config.TableFilter) > 0 {
-		for _, tableFilter := range config.TableFilter {
-			args = append(args, "--filter", tableFilter)
-		}
-	} else {
-		args = append(args, defaultTableFilterOptions...)
-	}
 	return args
 }
 
 // ConstructBRGlobalOptionsForRestore constructs BR global options for restore.
 func ConstructBRGlobalOptionsForRestore(restore *v1alpha1.Restore) ([]string, error) {
 	var args []string
-	config := restore.Spec.BR
-	if config == nil {
+	config := restore.Spec
```

Comment:
I think we can extract the common part of constructing the command line parameters from this function and reuse with `ConstructBRGlobalOptionsForBackup`